assembly/src/parsers/serde/deserialization.rs

```diff
@@ -373,14 +373,56 @@ impl Deserializable for Instruction {
             OpCode::CDropW => Ok(Instruction::CDropW),
 
             // ----- input / output operations --------------------------------------------------------
-            OpCode::PushConstants => {
+            OpCode::PushU8 => Ok(Instruction::PushU8(bytes.read_u8()?)),
+            OpCode::PushU16 => Ok(Instruction::PushU16(bytes.read_u16()?)),
+            OpCode::PushU32 => Ok(Instruction::PushU32(bytes.read_u32()?)),
+            OpCode::PushFelt => Ok(Instruction::PushFelt(bytes.read_felt()?)),
+            OpCode::PushWord => {
+                let mut constants = Vec::new();
+                for _ in 0..4 {
+                    constants.push(bytes.read_felt()?);
+                }
+                Ok(Instruction::PushWord(
+                    constants.try_into().expect("Invalid constatnts length"),
+                ))
+            }
+            OpCode::PushU8List => {
                 let mut constants = Vec::new();
                 let length = bytes.read_u8()?;
+                validate_param_number(length)?;
+                for _ in 0..length {
+                    constants.push(bytes.read_u8()?);
+                }
+                Ok(Instruction::PushU8List(constants))
+            }
+            OpCode::PushU16List => {
+                let mut constants = Vec::new();
+                let length = bytes.read_u8()?;
+                validate_param_number(length)?;
```

Comment:
Since the number of parameters is always encoded as `u8`, why not combine these functions into something like:
```Rust
fn parse_num_push_params(bytes: &mut ByteReader) -> Result<u8, SerializationError>
```